### ISSUE TITLE
fmq1.py

### ISSUE THREAD

**Issue by Kim-Q** [Kim-Q/Titanic-Machine-Learning-from-Disaster#1]
#!/usr/bin/python
# -*- coding: utf-8 -*-
#coding=utf-8

import csv
import pandas as pd
import numpy as np
import os
import matplotlib.pyplot as plt
from sklearn import tree
from sklearn import preprocessing as preprocessing
from math import log
from sklearn.linear_model import LogisticRegression
import operator

path='D:\DeepLearning\Titanic Dataset EDA'
TestData=pd.read_csv(path+'\\test.csv')
TrainData=pd.read_csv(path+'\\train.csv')

#TrainSurvived=TrainData[TrainData.Survived==1]
Sex_mapping={
    'female':1,
    'male':0
}
TrainData['Sex']=TrainData['Sex'].map(Sex_mapping)
TestData['Sex']=TestData['Sex'].map(Sex_mapping)
#TrainData.df['PassengerId','Survived','Pclass','Sex','Age','Fare']
#TrainData.dropna(
#    inplace=True
#)
TrainData['Age'].fillna(TrainData['Age'].mean(),inplace=True)
TestData['Age'].fillna(TestData['Age'].mean(),inplace=True)

#print(np.any(TrainData.isnull())==True)

#TrainData.drop(['PassengerId','Name','SibSp','Parch','Ticket','Cabin','Embarked'],axis=1,inplace=True)
#TestData.drop(['PassengerId','Name','SibSp','Parch','Ticket','Cabin','Embarked'],axis=1,inplace=True)

"""
scaler=preprocessing.StandardScaler()
age_scale_param=scaler.fit(TrainData['Age'])
TrainData['Age']=scaler.fit_transform(TrainData['Age'],age_scale_param)
fare_scale_param=scaler.fit(TrainData['Fare'])
TrainData['Fare']=scaler.fit_transform(TrainData['Fare'],fare_scale_param)
"""
#print(TrainData)
#print(TestData)

train_df=TrainData.filter(regex='Survived|Pclass|Sex|Age|Fare')
train_np=train_df.as_matrix
#print(train_np)

clf=LogisticRegression()
print clf
y=train_np[0:891,0]
X=train_np[0:891,1:4]
clf.fit(X,y)
clf

""" def calcShannonEnt(dataset):
    numEntris=891
    labelcounts={'1':0,'0':0}                                                     
    for featVec in dataset:
        currentlabel=featVec[1]                                           
        if currentlabel not in labelcounts.keys():                       
            labelcounts[currentlabel]=0
        labelcounts[currentlabel]+=1
    shannonEnt=0.0
    for key in labelcounts:
        prob=(float)(labelcounts[key]/numEntris)
        shannonEnt-=prob*log(prob,2)
    return shannonEnt

print(calcShannonEnt(TrainData)) """
